src/emqx_slow_subs/emqx_moving_average.erl

```diff
@@ -0,0 +1,90 @@
+%%--------------------------------------------------------------------
+%% Copyright (c) 2021 EMQ Technologies Co., Ltd. All Rights Reserved.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%--------------------------------------------------------------------
+
+%% @see https://en.wikipedia.org/wiki/Moving_average
+
+-module(emqx_moving_average).
+
+%% API
+-export([new/0, new/1, new/2, update/2]).
+
+-type type() :: cumulative
+              | exponential.
+
+-type ema() :: #{ type := exponential
+                , average := 0 | float()
+                , coefficient := float()
+                }.
+
+-type cma() :: #{ type := cumulative
+                , average := 0 | float()
+                , count := non_neg_integer()
+                }.
+
+-type moving_average() :: ema()
+                        | cma().
+
+-define(DEF_EMA_ARG, #{period => 10}).
+-define(DEF_AVG_TYPE, cumulative).
```

Comment:
default was exponential the last time, why change to cumulative ?
in cumulative, old values have long time impact on new values.